utils/build-script

```diff
@@ -1473,6 +1514,191 @@ details of the setups of other systems or automated environments.""")
+    #
+    # FIXME: Allow Android host tests to be enabled/disabled by the build
+    # script.
+    platforms_to_skip_test.add(StdlibDeploymentTarget.Android)
+
+    def get_options_for_target(host):
+        def should_build_stdlib_target(stdlib_target):
+            # If building all targets, this is true.
+            if "all" in args.build_stdlib_deployment_targets:
+                return True
+
+            # Otherwise, only build in the restricted list.
+            if stdlib_target in args.build_stdlib_deployment_targets:
+                return True
+
+            # For cross-tools hosts, always allow their one-and-only
+            # stdlib-target to build.
+            if host != args.host_target and host == stdlib_target:
+                return True
+
+        if host == args.host_target:
+            stdlib_targets = args.stdlib_deployment_targets
+        else:
+            stdlib_targets = [host]
+
+        swift_stdlib_targets = []
+        swift_sdks = set()
+        swift_benchmark_targets = []
+        swift_run_benchmark_targets = []
+        swift_test_targets = []
+        for stdlib_target_name in stdlib_targets:
+            # Get the target object.
+            stdlib_target = StdlibDeploymentTarget.get_target_for_name(
+                stdlib_target_name)
+            if stdlib_target is None:
+                diagnostics.fatal("unknown target: %r" % (stdlib_target_name,))
+
+            # Add the SDK to use.
+            swift_sdks.add(stdlib_target.platform.sdk_name)
+
+            # If we aren't building this target, do nothing else.
+            #
+            # FIXME: This matches build-script-impl behavior, but why add the
+            # SDK? Is that right?
+            if not should_build_stdlib_target(stdlib_target):
+                continue
+
+            # Compute build and test information.
+            build = stdlib_target.platform not in platforms_to_skip_build
+            test = stdlib_target.platform not in platforms_to_skip_test
+            test_host_only = None
+            build_benchmark = build and stdlib_target.supports_benchmark
+            # FIXME: Note, `build-script-impl` computed a property here
+            # w.r.t. testing, but it was actually unused.
+
+            # As a special case, Darwin embedded non-simulator platforms have
+            # different behavior for the test condition.
+            if stdlib_target.platform.is_darwin and \
+               stdlib_target.platform.is_embedded and \
+               not stdlib_target.platform.is_simulator:
+                if stdlib_target.platform not in platforms_to_skip_test_host:
+                    test_host_only = True
+                    test = True
+                else:
+                    test = False
+
+            name = stdlib_target.name
+            if build:
+                # FIXME: This logic doesn't make much sense, but
+                # build-script-impl had no comments.
```

Comment:
It builds `swift-test-stdlib-*` if you asked to build.  It builds `swift-stdlib-*` instead if you specify `--build-swift-stdlib-unittest-extra=1`.

Are you saying that it is confusing that it is building something with "test" in it even though we are not testing?  I can see that.  The idea was that the `swift-test-stdlib` set is what developers usually need to develop the compiler and run most of the tests.  So even though you are not requesting to run the tests, that is usually the reason why this target is being built -- with an intention of running the tests, either as a part of build-script, or by manually invoking `ninja`.

I'm open to better suggestions -- I agree this is very ad-hoc.